eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -143,42 +218,130 @@ object Transactions {
+      val inputsMap = extraUtxos + (input.outPoint -> input.txOut)
+      val allInputsProvided = tx.txIn.forall(txIn => inputsMap.contains(txIn.outPoint))
+      val witnessesOk = Try(Transaction.correctlySpends(tx, inputsMap, ScriptFlags.STANDARD_SCRIPT_VERIFY_FLAGS)).isSuccess
+      allInputsProvided && witnessesOk
     }
   }
-  // @formatter:on
 
-  sealed trait ReplaceableTransactionWithInputInfo extends TransactionWithInputInfo {
-    /** Block before which the transaction must be confirmed. */
-    def confirmationTarget: ConfirmationTarget
+  /**
+   * Transactions spending the channel funding output: [[CommitTx]], [[SpliceTx]] and [[ClosingTx]].
+   * Those transactions always require two signatures, one from each channel participant.
+   */
+  sealed trait ChannelSpendTransaction extends TransactionWithInputInfo {
+    /** Sign the channel's 2-of-2 funding output when using a [[SegwitV0CommitmentFormat]]. */
+    def sign(localFundingKey: PrivateKey, remoteFundingPubkey: PublicKey, extraUtxos: Map[OutPoint, TxOut]): ChannelSpendSignature.IndividualSignature = {
+      val redeemScript = Script.write(Scripts.multiSig2of2(localFundingKey.publicKey, remoteFundingPubkey))
+      val sig = sign(localFundingKey, SIGHASH_ALL, RedeemInfo.P2wsh(redeemScript), extraUtxos)
+      ChannelSpendSignature.IndividualSignature(sig)
+    }
+
+    /** Create a partial transaction for the channel's musig2 funding output when using a [[TaprootCommitmentFormat]]. */
+    def partialSign(localFundingKey: PrivateKey, remoteFundingPubkey: PublicKey, extraUtxos: Map[OutPoint, TxOut]): ChannelSpendSignature.PartialSignatureWithNonce = {
+      ???
+    }
+
+    /** Verify a signature received from the remote channel participant. */
+    def checkRemoteSig(localFundingPubkey: PublicKey, remoteFundingPubkey: PublicKey, remoteSig: ChannelSpendSignature): Boolean = {
+      remoteSig match {
+        case IndividualSignature(remoteSig) =>
+          val redeemScript = Script.write(Scripts.multiSig2of2(localFundingPubkey, remoteFundingPubkey))
+          checkSig(remoteSig, remoteFundingPubkey, SIGHASH_ALL, RedeemInfo.P2wsh(redeemScript))
+        case PartialSignatureWithNonce(_, _) => ???
+      }
+    }
+
+    /** Aggregate local and remote channel spending signatures for a [[SegwitV0CommitmentFormat]]. */
+    def aggregateSigs(localFundingPubkey: PublicKey, remoteFundingPubkey: PublicKey, localSig: IndividualSignature, remoteSig: IndividualSignature): Transaction = {
+      val witness = Scripts.witness2of2(localSig.sig, remoteSig.sig, localFundingPubkey, remoteFundingPubkey)
+      tx.updateWitness(inputIndex, witness)
+    }
+
+    /** Aggregate local and remote channel spending partial signatures for a [[TaprootCommitmentFormat]]. */
+    def aggregateSigs(localFundingPubkey: PublicKey, remoteFundingPubkey: PublicKey, localSig: PartialSignatureWithNonce, remoteSig: PartialSignatureWithNonce): Transaction = {
```

Comment:
Same as with `partialSign`:
```suggestion
    def partialSign(localFundingKey: PrivateKey, remoteFundingPubkey: PublicKey, extraUtxos: Map[OutPoint, TxOut], localNonce: (SecretNonce, IndividualNonce), publicNonces: Seq[IndividualNonce]): Either[Throwable, ChannelSpendSignature.PartialSignatureWithNonce] = {
```